eval.go

```diff
@@ -269,6 +269,18 @@ func (e *setExpr) eval(app *app, args []string) {
 			}
 		}
 		gOpts.info = toks
+	case "locale":
+		localeStr := e.val
+		if gOpts.locale != localeStr {
+			if localeStr != localeStrDisable {
+				_, err = getLocaleTag(localeStr)
+			}
+			if err == nil {
```

Comment:
The `err` in this section of the code generally means that the syntax of the `set` command is wrong and should be reported to users. So either:

1. Display this error using `app.ui.echoerr`
2. Suppress the error and just fall back to the default sort method, in which case you only need
   ```go
   gOpts.locale = localeStr
   app.nav.sort()
   app.ui.sort()
   ```

Same for `setlocal` below.

Reply:
If I get it right, when I want to display error message to user, I should write this instead right?

```go
if localeStr != localeStrDisable {
	if _, err = getLocaleTag(localeStr); err != nil {
		app.ui.echoerr(/* some message */)
	}
}
// ...
```

Reply:
I also want to confirm if early return in a case branch is OK. If yes, I'd like to change outermost `if` in this branch into an early return, in my opinion that would look better.

From:

```go
localeStr := e.val
if gOpts.locale != localeStr {
    // ...
}
```

To:

```go
localeStr := e.val
if gOpts.locale == localeStr {
    return
}
// ...
```

Reply:
Early return should be fine. You can use something like below:

```go
localeStr := e.val
if localeStr != localeStrDisable {
    if _, err = getLocaleTag(localeStr); err != nil {
        app.ui.echoerrf("locale: %s", err.Error())
        return
    }
}
gOpts.locale = localeStr
app.nav.sort()
app.ui.sort()
```

I think there's not much benefit to checking if the new value is different from the old setting, I expect users would set this only once, especially given that `setlocal` can be used for specific directories.

Reply:
OK, on it.